Back-port from OpenAMP project

arnopo opened the pull request:
backport of 2 patches pushed on https://github.com/OpenAMP/open-amp to fix compilation issues on zephyr project.

https://github.com/OpenAMP/open-amp/pull/198
https://github.com/OpenAMP/open-amp/pull/199

galak reviewed (commented):
Do we want to call this a backport or just sync with SHA: dafae9e2e2b5e80b37aec37e6c3f9ad991e2dfd0

arnopo commented:
> Do we want to call this a backport or just sync with SHA: dafae9e2e2b5e80b37aec37e6c3f9ad991e2dfd0

it is a cherry-picking of the patches sent by @SebastianBoe , so not a strict sync. Do you expect a resync on OpenAMP master?

galak commented:
> it is a cherry-picking of the patches sent by @SebastianBoe , so not a strict sync. Do you expect a resync on OpenAMP master?

I'm saying maybe we should just re-sync on master instead of backporting the patches.  I don't see any particular reason not to sync w/master.

arnopo commented:
> > it is a cherry-picking of the patches sent by @SebastianBoe , so not a strict sync. Do you expect a resync on OpenAMP master?
> 
> I'm saying maybe we should just re-sync on master instead of backporting the patches. I don't see any particular reason not to sync w/master.

ok, i will do it in this way for  libmetal also

arnopo committed:
Sync with openamp project sha1 2af66b535

Synchronize code with sha1 2af66b535848b920e23d5522f924ae0128780325
to integrate fix for cmake3.17 support

Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@st.com>
